Detections/MultipleDataSources/WSLMalwareCorrelation.yaml

```diff
@@ -0,0 +1,181 @@
+    | extend timestamp = TimeGenerated, DNSName = RemoteUrl, IPCustomEntity = RemoteIP, HostCustomEntity = DeviceName,  AlertDetail =  'Alert for  IOCs related to Windows/ELF malware'
+    ),
+    (WindowsFirewall
+    | where SourceIP in (IPList) or DestinationIP in (IPList) 
+    | project TimeGenerated, Computer, CommunicationDirection, SourceIP, DestinationIP, SourcePort, DestinationPort
+    | extend IPMatch = case( SourceIP in (IPList), "SourceIP", DestinationIP in (IPList), "DestinationIP", "None"), AlertDetail =  'Alert for  IOCs related to Windows/ELF malware'
+    | extend timestamp = TimeGenerated, HostCustomEntity = Computer , IPCustomEntity = case(IPMatch == "SourceIP", SourceIP, IPMatch == "DestinationIP", DestinationIP, "None")
+    ),
+    (AzureDiagnostics
+    | where ResourceType == "AZUREFIREWALLS"
+    | where Category == "AzureFirewallDnsProxy"
+    | project TimeGenerated,Resource, msg_s
+    | parse msg_s with "DNS Request: " ClientIP ":" ClientPort " - " QueryID " " Request_Type " " Request_Class " " Request_Name ". " Request_Protocol " " Request_Size " " EDNSO_DO " " EDNS0_Buffersize " " Responce_Code " " Responce_Flags " " Responce_Size " " Response_Duration
+    | where  ClientIP in (IPList)
+    | extend timestamp = TimeGenerated, DNSName = Request_Name, IPCustomEntity = ClientIP, AlertDetail = 'Alert for  IOCs related to Windows/ELF malware'
+    ),
+    (AzureDiagnostics 
+    | where ResourceType == "AZUREFIREWALLS"
+    | where Category == "AzureFirewallApplicationRule"
+    | project TimeGenerated,Resource, msg_s
+    | parse msg_s with Protocol 'request from ' SourceHost ':' SourcePort 'to ' DestinationHost ':' DestinationPort '. Action:' Action
+    | where isnotempty(DestinationHost) 
+    | where SourceHost has "IPList"
+    | extend timestamp = TimeGenerated, DNSName = DestinationHost, IPCustomEntity = SourceHost, AlertDetail = 'Alert for  IOCs related to Windows/ELF malware'
+    ),
+    (DeviceFileEvents
+    | project TimeGenerated, ActionType, DeviceId, DeviceName, InitiatingProcessAccountDomain, InitiatingProcessAccountName, InitiatingProcessCommandLine, InitiatingProcessFolderPath, InitiatingProcessId, InitiatingProcessParentFileName, InitiatingProcessFileName, RequestAccountName, RequestSourceIP, InitiatingProcessSHA256
+    | extend Account = RequestAccountName, Computer = DeviceName, IPAddress = RequestSourceIP, CommandLine = InitiatingProcessCommandLine, FileHash = InitiatingProcessSHA256, AlertDetail = 'Alert for  IOCs related to Windows/ELF malware'
+    | extend timestamp = TimeGenerated, HostCustomEntity = Computer , AccountCustomEntity = Account, ProcessCustomEntity = InitiatingProcessFileName, FileHashCustomEntity = FileHash
+    | where FileHash has "sha256hashes"
```

Comment:
should be `in (sha256Hashes)` or `has_any (sha256Hashes)` instead of `has "sha256hashes"`